app/views/handlers/individual_response.py

```diff
@@ -150,6 +150,24 @@ def handle_get(self):
             self.individual_section_id
         )
 
+        self._list_name = self._schema.get_individual_response_list()
+        list_model = self._questionnaire_store.list_store[self._list_name]
+
+        if self._list_item_id:
+            next_location_url = url_for(
+                ".individual_response_how",
+                list_item_id=self._list_item_id,
+                journey=self._request_args.get("journey"),
+            )
+        elif len(list_model.non_primary_people) == 1:
+            next_location_url = url_for(
+                ".individual_response_how",
+                list_item_id=list_model.non_primary_people[0],
+                journey="hub",
+            )
+        else:
+            next_location_url = url_for(".individual_response_who", journey="hub")
+
```

Comment:
There is a lot going on in this method now. I think we should create `get_previous_location_url` and `get_next_location_url` methods, and call them from here.